statsd/statsd.go

```diff
@@ -681,15 +685,23 @@ func (c *Client) sendToAggregator(mType metricType, name string, value float64,
 }
 
 // Gauge measures the value of a metric at a particular time.
-func (c *Client) Gauge(name string, value float64, tags []string, rate float64) error {
+func (c *Client) Gauge(name string, value float64, tags []string, rate float64, parameters ...Parameter) error {
 	if c == nil {
 		return ErrNoClient
 	}
 	atomic.AddUint64(&c.telemetry.totalMetricsGauge, 1)
 	if c.agg != nil {
 		return c.agg.gauge(name, value, tags)
```

Comment:
A caveat I forgot to mention... 

The go client supports aggregating metrics. When aggregating, we need to take into account the cardinality, so metrics with different cardinality settings are not aggregated together. I think this will largely involve passing the cardinality through and including it as a part of the context in the GetContextAndTags. Then we need to make sure the cardinality is included when the metric gets flushed.